ampd/src/handlers/multisig.rs

```diff
@@ -353,7 +388,44 @@ mod test {
     }
 
     #[tokio::test]
-    async fn should_not_handle_event_if_multisig_address_does_not_match() {
+    async fn should_not_handle_wrong_event_if_multisig_address_does_not_match() {
+        let mut client = MockEcdsaClient::new();
+        client
+            .expect_sign()
+            .returning(move |_, _, _| Err(Report::from(tofnd::error::Error::SignFailed)));
+
+        let handler = get_handler(
+            rand_account(),
+            rand_account(),
+            SharableEcdsaClient::new(client),
+            100u64,
+        );
+
+        assert!(handler.handle(&wrong_signing_started_event()).await.is_ok());
+    }
+
+    #[tokio::test]
+    async fn should_error_on_wrong_event_if_multisig_address_does_match() {
+        let mut client = MockEcdsaClient::new();
+        client
+            .expect_sign()
+            .returning(move |_, _, _| Err(Report::from(tofnd::error::Error::SignFailed)));
+
+        let handler = get_handler(
+            rand_account(),
+            TMAddress::from(MULTISIG_ADDRESS.parse::<AccountId>().unwrap()),
+            SharableEcdsaClient::new(client),
+            100u64,
+        );
+
+        assert!(handler
+            .handle(&wrong_signing_started_event())
+            .await
+            .is_err());
+    }
```

Comment:
looking at these tests I do not understand what is being tested if I don't drill into the code. For example, the client always returns an error, is that what the assert catches? With the `MULTISIG_ADDRESS` const call buried in `wrong_signing_started_event` I have no way of knowing that it's actually matching the address you give to the handler

Reply:
tried to clean this up and make it more clear